### PR TITLE
support auto rollback when report exception in transaction block.

### PR DESCRIPTION
**Pull request by jingshanglu**
[design detail](https://u01f1kqxrl.feishu.cn/docs/doccntaO4s7K5I5FbML5UAponJe#)

Changes proposed in this pull request:
- support auto rollback when report exception in transaction block for postgresql and opengauss

Test
[jdbc test case](https://github.com/SphereEx/auto-test/tree/master/src/main/java/com/sphereex/cases/transaction/shardingjdbc)

### CODE REVIEW

**Comment by terrymanu** [comment]
> [design detail](https://u01f1kqxrl.feishu.cn/docs/doccntaO4s7K5I5FbML5UAponJe#)
> 
> Changes proposed in this pull request:
> 
> * support auto rollback when report exception in transaction block for postgresql and opengauss

@jingshanglu Hi, could you give the English version on GitHub?

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/command/CommandExecutorTask.java [hunk 1]
```diff
@@ -107,6 +110,12 @@ private boolean executeCommand(final ChannelHandlerContext context, final Packet
         return databaseProtocolFrontendEngine.getFrontendContext().isFlushForPerCommandPacket();
     }
     
+    private void setRollbackOnly() {
+        if (databaseProtocolFrontendEngine.getDatabaseType().equals("PostgreSQL") || "openGauss".equals(databaseProtocolFrontendEngine.getDatabaseType())) {
+            connectionSession.getTransactionStatus().setRollbackOnly(true);
+        }
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/command/CommandExecutorTask.java:116]: Please use SPI

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/ShardingSphereTransactionManager.java [hunk 2]
```diff
@@ -80,6 +80,13 @@
      */
     void commit();
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/ShardingSphereTransactionManager.java:81]: Is `commit` still useful?

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by jingshanglu** [comment]
> > [design detail](https://u01f1kqxrl.feishu.cn/docs/doccntaO4s7K5I5FbML5UAponJe#)
> > Changes proposed in this pull request:
> > 
> > * support auto rollback when report exception in transaction block for postgresql and opengauss
> 
> @jingshanglu Hi, could you give the English version on GitHub?

ok

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/ShardingSphereTransactionManager.java (hunk 2 above)
Comment (jingshanglu) [reply shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/ShardingSphereTransactionManager.java:81]: sharding-jdbc use it.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**